accounting/accounting/doctype/stock_entry/test_stock_entry.py

```diff
@@ -0,0 +1,164 @@
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.incoming_warehouse_name = "".join(random.choices(string.ascii_letters, k=10))
+        self.outgoing_warehouse_name = "".join(random.choices(string.ascii_letters, k=10))
+        self.main_warehouse_name = "".join(random.choices(string.ascii_letters, k=10))
+
+    def setUp(self):
+        frappe.set_user("Administrator")
+
+        # Create some items
+        name = "".join(random.choices(string.ascii_letters, k=10))
+        frappe.get_doc(
+            {
+                "doctype": "Item",
+                "item_name": name,
+            }
+        ).insert()
+        frappe.get_doc(
+            {
+                "doctype": "Item",
+                "item_name": name[::-1],
+            }
+        ).insert()
+
+        # Create some warehouses
+        frappe.get_doc(
+            {
+                "doctype": "Warehouse",
+                "warehouse_name": self.incoming_warehouse_name,
+                "address": "".join(random.choices(string.ascii_letters, k=10)),
+            }
+        ).insert()
+        frappe.get_doc(
+            {
+                "doctype": "Warehouse",
+                "warehouse_name": self.outgoing_warehouse_name,
+                "address": "".join(random.choices(string.ascii_letters, k=10)),
+            }
+        ).insert()
+        frappe.get_doc(
+            {
+                "doctype": "Warehouse",
+                "warehouse_name": self.main_warehouse_name,
+                "address": "".join(random.choices(string.ascii_letters, k=10)),
+            }
+        )
+
+    def tearDown(self):
+        frappe.db.rollback()
+
+    def test_create_stock_entry_without_mandatory_fields(self):
+        frappe.set_user("Administrator")
+        doc = frappe.get_doc(
+            {
+                "doctype": "Stock Entry",
+            }
+        )
+        with self.assertRaises(frappe.exceptions.ValidationError):
+            doc.insert()
+
+    def test_create_stock_entry(self):
+        frappe.set_user("Administrator")
+        item_names = frappe.db.get_all("Item", fields=["name"])
+
+        items = []
+        for item in item_names:
+            items.append(
+                {
+                    "item": item["name"],
+                    "quantity": random.randint(10, 15),
+                    "rate": random.randint(100, 1000),
+                }
+            )
+
+        # Test receiving items
+        doc = frappe.get_doc(
+            {
+                "doctype": "Stock Entry",
+                "entry_type": "Receipt",
+                "target_warehouse": self.incoming_warehouse_name,
+                "items": items,
+            }
+        )
+        doc.insert()
+        created_stock_entry = frappe.get_doc("Stock Entry", doc.name)
+        self.assertEqual(created_stock_entry.entry_type, "Receipt")
+
+        # Check ledger entries
+        for item in items:
+            data = frappe.db.get(
```

Comment:
```suggestion
            data = frappe.db.get_value(
```


get is kinda deprecated.